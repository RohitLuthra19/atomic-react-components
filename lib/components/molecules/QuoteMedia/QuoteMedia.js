// @flow
/**
 *
 * QuoteMedia
 *
 */
import React from 'react';
import styled from 'styled-components';

import type { Node } from 'react';
import styles from './QuoteMedia.style';
import type { Props } from './types';

import Heading from '../../atoms/Heading/Heading';

const QuoteMedia = ({
  className,
  title,
  quoteContent,
  HeadingType,
  variant,
  children,
}: Props): Node => (
  <div className={className}>
    <Heading HeadingType={HeadingType} className="title">
      {title}
    </Heading>
    <div className={`${className} ${variant}`}>
      {quoteContent && quoteContent()}
      {children && <div className="media">{children}</div>}
    </div>
  </div>
);

QuoteMedia.defaultProps = {
  className: 'quoteMedia',
  title: 'Anonymous',
  variant: ['stacked', 'column'],
};

export default styled(QuoteMedia)`
  ${styles};
`;

export { QuoteMedia as QuoteMediaVanilla };
