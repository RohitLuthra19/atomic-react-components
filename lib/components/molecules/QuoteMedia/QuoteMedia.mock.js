import Blockquote from '../../atoms/Blockquote/Blockquote';

const defaultConfig = {
  className: 'blockquote-container',
  quoteClassName: 'blockquote',
  HeadingType: 'h2',
  title: 'Lorem ipsum dollor',
  quoteAuthorProp: 'Lorem ipsum',
  quoteProp: 'Lorem ipsum dollor sit amet',
  quoteContent: (quoteClassName, quoteAuthorProp, quoteProp) => (
    <Blockquote
      className={quoteClassName}
      quoteAuthorProp={quoteAuthorProp}
      quoteProp={quoteProp}
    />
  ),
};

export default defaultConfig;
