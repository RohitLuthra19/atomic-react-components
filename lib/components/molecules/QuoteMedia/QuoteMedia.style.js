import { css } from 'styled-components';

const styles = css`
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  .title {
    ${props => `
        ${props.theme.typography.FONT_H2};
        padding: ${props.theme.spacing.unit2};
    `};
    grid-column: 1/-1;
  }

  .media {
    ${props => `
        padding: ${props.theme.spacing.unit2};
  `};
  }

  .blockquote {
    position: relative;
    ${props => `
      padding: ${props.theme.spacing.unit1} ${props.theme.spacing.unit9};
    `};
    &__content {
      ${props => `
        ${props.theme.typography.FONT_H2};
        padding: ${props.theme.spacing.unit1}
      `};
    }
    &__author {
      text-align: right;
      ${props => `
      ${props.theme.typography.FONT_H3};
      padding: ${props.theme.spacing.unit1}
    `};
    }
    &:before {
      display: block;
      height: 0;
      content: '”';
      position: absolute;
      top: 0;
      font-family: var(--font-family-display);
      line-height: 1;
      ${props => `
      left: ${props.theme.spacing.unit2}
      font-size: ${props.theme.spacing.unit16}
      color: ${props.theme.colors.TEXT_LIGHT}
    `};
    }
  }
  ${props => (props.inheritedStyles ? props.inheritedStyles : '')};
`;

export default styles;
