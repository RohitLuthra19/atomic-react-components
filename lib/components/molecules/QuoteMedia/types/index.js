// @flow
import type { Node } from 'react';

export type Props = {
  className?: string,
  title?: string,
  author: string,
  quoteContent: string,
  childClassName?: string,
  children: Node,
};
