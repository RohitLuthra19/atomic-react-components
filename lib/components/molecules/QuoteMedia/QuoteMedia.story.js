import React from 'react';
import { storiesOf } from '@storybook/react';
// import { action } from '@storybook/addon-actions';
import defaultConfig from './QuoteMedia.mock';

// Import Styled Component to showcase variations
import QuoteMedia, { QuoteMediaVanilla } from '.';

// TODO: indentation is not working fine in editor. Need to check for .hbs files
// $FlowFixMe eslint-disable-next-line extra-rules/no-commented-out-code
storiesOf('Molecules', module)
  .addParameters({
    jest: ['QuoteMedia', 'QuoteMediaVanilla'],
  })
  .add('QuoteMedia', () => (
    <QuoteMediaVanilla
      {...defaultConfig}
      className={`hide-default-sample ${defaultConfig.className}`}
    />
  ))
  .add('QuoteMedia', () => <QuoteMedia {...defaultConfig} />);
