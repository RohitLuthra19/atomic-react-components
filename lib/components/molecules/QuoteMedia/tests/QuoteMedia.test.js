import React from 'react';
import { shallow } from 'enzyme';
import { QuoteMediaVanilla } from '../QuoteMedia';
import defaultQuoteMedia from '../QuoteMedia.mock';

describe('<QuoteMedia />', () => {
  let QuoteMediaComponent = '';
  beforeEach(() => {
    QuoteMediaComponent = shallow(<QuoteMediaVanilla {...defaultQuoteMedia} />);
  });
  it('should render header', () => {
    expect(QuoteMediaComponent.find('h2')).toBeTruthy();
  });
  it.only('check heading value passed in', () => {
    expect(QuoteMediaComponent.find('.title').text()).toBe('Lorem ipsum dollor');
  });
  it('check props passed in', () => {
    expect(QuoteMediaComponent.props().quoteContent).not.toBeNull();
  });
  test('should render correctly', () => {
    expect(QuoteMediaComponent).toMatchSnapshot();
  });
});
