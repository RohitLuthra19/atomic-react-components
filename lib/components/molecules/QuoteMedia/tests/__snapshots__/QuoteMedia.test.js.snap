// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuoteMedia /> should render correctly 1`] = `
<div
  className="blockquote-container"
>
  <Heading
    HeadingType="h2"
    className="title"
  >
    Lorem ipsum dollor
  </Heading>
  <Blockquote
    className="blockquote"
    quoteAuthorProp="Lorem ipsum"
    quoteProp="Lorem ipsum dollor sit amet"
  />
  <div
    className="media"
  />
</div>
`;
